### PR TITLE
PowerShell command line completion

### PR DESCRIPTION
**Pull request by stax76**
As the title says, PowerShell command line completion. Tested on Windows and Linux.

### CODE REVIEW

File: etc/mpv-powershell-completion.ps1 [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+
+Register-ArgumentCompleter -Native -CommandName mpv -ScriptBlock {
+    param($commandName, $wordToComplete, $cursorPosition)
+
+    Get-Completion $cursorPosition "$wordToComplete" "$commandName" | ForEach-Object {
+        [System.Management.Automation.CompletionResult]::new($_, $_, 'ParameterValue', $_)
+    }
+}
+
+Register-ArgumentCompleter -Native -CommandName mpvnet -ScriptBlock {
```
Comment (hooke007) [inline etc/mpv-powershell-completion.ps1:281]: Not sure if all one would like it...

File: etc/mpv-powershell-completion.ps1 [hunk 2]
```diff
@@ -0,0 +1,287 @@
+#!/bin/pwsh
+
+# PowerShell command line completion for the mpv/mpv.net media player.
```
Comment (hooke007) [inline etc/mpv-powershell-completion.ps1:3]: IMO, it should be mpv only here.

**Comment by hooke007** [comment]
related https://github.com/mpv-player/mpv/issues/12957

File: etc/mpv-powershell-completion.ps1 [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+$Options = New-Object Collections.Generic.List[Object]
+$UpdateOptions = @('vo', 'ao', 'profile', 'hwdec', 'audio-device', 'scale', 'tscale',
```
Comment (hooke007) [inline etc/mpv-powershell-completion.ps1:25]: Any reason why only these options?

**Comment by hooke007** [comment]
Also, I think the 'uninstaller' should also be provided.

File: etc/mpv-powershell-completion.ps1 (hunk 3 above)
Comment (stax76) [reply etc/mpv-powershell-completion.ps1:25]: These options have `=help` support, which the script uses to extract the choices by running the executable.

Comment (hooke007) [reply etc/mpv-powershell-completion.ps1:25]: But more options also support `help`...

File: etc/mpv-powershell-completion.ps1 [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+            }
+
+            $table["value"] = $value
+        }
+        else
+        {
+            $table["name"] = $line.Substring(2)
+        }
+
+        if ($value.StartsWith("Choices:"))
```
Comment (kasper93) [inline etc/mpv-powershell-completion.ps1:67]: `$value` is not initialized when we don't go into `if ($line.Contains(" "))` block

File: etc/mpv-powershell-completion.ps1 [hunk 5]
```diff
@@ -0,0 +1,287 @@
+#!/bin/pwsh
```
Comment (kasper93) [inline etc/mpv-powershell-completion.ps1:1]: Not sure it need shebang, it suppose to be sourced anyway, not executed.

File: etc/mpv-powershell-completion.ps1 (hunk 4 above)
Comment (stax76) [reply etc/mpv-powershell-completion.ps1:67]: I'll fix it. There is also often double quotes where single quotes would be better. Shebang can be removed.

Problem is I'm really not so good with git, for instance I've forgotten to make a branch for the PR, what do I do now?

Comment (kasper93) [reply etc/mpv-powershell-completion.ps1:67]: Just use the `master` branch. Once this PR is merged/closed you will be able to reuse this branch like nothing happens.  `git commit --amend` your changes. Don't forget to change the commit name to `powershell-completion: add PowerShell command line completion` or similar that matches commit naming requirements of mpv.

Comment (stax76) [reply etc/mpv-powershell-completion.ps1]: I wasn't able to squash the commits, then created a new pull request here:

https://github.com/mpv-player/mpv/pull/12967/files

**Pull request closed without merging**